### PR TITLE
[ISSUE #4114] Add threadPool in DefaultMQAdminExtImpl and execute queryConsumeTimeSpan concurrently

### PR DESCRIPTION
**Pull request by MatrixHB**
To solve the problem of [ISSUE #4114](https://github.com/apache/rocketmq/issues/4114)

execute `queryConsumeTimeSpan` and `examineTopicStats` concurrently, which are used when query messages in certain time span.

### CODE REVIEW

File: tools/src/main/java/org/apache/rocketmq/tools/admin/DefaultMQAdminExtImpl.java [hunk 1]
```diff
@@ -147,6 +157,9 @@ public void start() throws MQClientException {
 
                 mqClientInstance.start();
 
+                executors = new ThreadPoolExecutor(10, 20, 1000 * 30, TimeUnit.MILLISECONDS,
+                    new ArrayBlockingQueue<>(100), new ThreadFactoryImpl("MQAdminThread_"), new ThreadPoolExecutor.CallerRunsPolicy());
```
Comment (dugenkui03) [inline tools/src/main/java/org/apache/rocketmq/tools/admin/DefaultMQAdminExtImpl.java:161]: I wonder why not 'CachedThreadPool', like `new ThreadPoolExecutor(0, Integer.MAX_VALUE, 1000 * 30, TimeUnit.MILLISECONDS...`.

Comment (MatrixHB) [reply tools/src/main/java/org/apache/rocketmq/tools/admin/DefaultMQAdminExtImpl.java:161]: Good suggestion. Actually this thread pool is idle most of the time...

**Comment by dugenkui03** [comment]
Hi, If this change is really accept by rocketmq-team. I suggest create a common method, which can do work concurrently and **throw the original Exception**. Details see review comment.

**Comment by MatrixHB** [comment]
@dugenkui03
 I‘ve just found that in the `5.0.0-beta` branch provided by rocketmq-team, there is already `queryConsumeTimeSpanConcurrent` method to solve my problem.
But the `queryConsumeTimeSpanConcurrent` added in 5.0.0-beta has some problems:
1、Using countDownLatch, `spanSet.addAll()` executed in each thread is not thread-safe, because spanSet is not copy-on-write
2、IMO, add concurrent method separately makes DefaultMQAdminExtImpl method-expansion, not so elegant.
3、The thread pool parameter setting can be optimized.

File: tools/src/main/java/org/apache/rocketmq/tools/admin/DefaultMQAdminExtImpl.java [hunk 2]
```diff
@@ -147,6 +157,8 @@ public void start() throws MQClientException {
 
                 mqClientInstance.start();
 
+                executors = Executors.newCachedThreadPool(new ThreadFactoryImpl("MQAdminThread_"));
```
Comment (lizhiboo) [inline tools/src/main/java/org/apache/rocketmq/tools/admin/DefaultMQAdminExtImpl.java:160]: IMO，it's better to specify pool core size, max size, length of waiting list and reject conduct.

**Comment by lwclover** [comment]
Wouldn't it be better to default to a thread pool?

**Pull request closed without merging**